target/linux/mediatek/dts/mt7981b-mercusys-mr80x-v3.dts

```diff
@@ -0,0 +1,352 @@
+
+/dts-v1/;
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+#include "mt7981.dtsi"
+/ {
+	model = "MediaTek MT7981 RFB";
+	compatible = "mercusys,mr80x-v3";
+	chosen {
+		bootargs = "console=ttyS0,115200n1 loglevel=8  \
+				earlycon=uart8250,mmio32,0x11002000 \
+				init=/etc/preinit";
+	};
+
+	memory {
+		reg = <0 0x40000000 0 0x10000000>;
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&pio 1 GPIO_ACTIVE_LOW>;
+			linux,input-type = <1>;
+		};
+	};
+	leds {
+		compatible = "gpio-leds";
+
+		orange {
+			label = "orange";
+			default-state = "on";
+			gpios = <&pio 4 GPIO_ACTIVE_HIGH>;
+		};
+		green {
+			label = "green";
+			default-state = "off";
+			gpios = <&pio 5 GPIO_ACTIVE_HIGH>;
+		};
+		wan {
+			label = "wan";
+			default-state = "off";
+			gpios = <&pio 9 GPIO_ACTIVE_LOW>;
+		};
+		lan1 {
+			label = "lan1";
+			default-state = "off";
+			gpios = <&pio 6 GPIO_ACTIVE_LOW>;
+		};
+		lan2 {
+			label = "lan2";
+			default-state = "off";
+			gpios = <&pio 7 GPIO_ACTIVE_LOW>;
+		};
+		lan3 {
+			label = "lan3";
+			default-state = "off";
+			gpios = <&pio 10 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	sound_wm8960 {
+		compatible = "mediatek,mt79xx-wm8960-machine";
+		mediatek,platform = <&afe>;
+		audio-routing = "Headphone", "HP_L",
+				"Headphone", "HP_R",
+				"LINPUT1", "AMIC",
+				"RINPUT1", "AMIC";
+		mediatek,audio-codec = <&wm8960>;
+		status = "disabled";
+	};
+
+	sound_si3218x {
+		compatible = "mediatek,mt79xx-si3218x-machine";
+		mediatek,platform = <&afe>;
+		mediatek,ext-codec = <&proslic_spi>;
+		status = "disabled";
+	};
+};
+
+&afe {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pcm_pins>;
+	status = "okay";
+};
+
+&i2c0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c_pins>;
+	status = "disabled";
+
+	wm8960: wm8960@1a {
+		compatible = "wlf,wm8960";
+		reg = <0x1a>;
+	};
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&watchdog {
+	status = "okay";
+};
+
+&eth {
+	status = "okay";
+
+	gmac0: mac@0 {
+		compatible = "mediatek,eth-mac";
+		reg = <0>;
+		phy-mode = "2500base-x";
+
+		fixed-link {
+			speed = <2500>;
+			full-duplex;
+			pause;
+		};
+	};
+
+	gmac1: mac@1 {
+		compatible = "mediatek,eth-mac";
+		reg = <1>;
+		phy-mode = "gmii";
+		nick-index = <4>;
+		phy-handle = <&phy0>;
+		phy-link {
+            speed = <1000>;
+            full-duplex;
+        };
+
+	};
+
+	mdio: mdio-bus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		phy5: phy@5 {
+			compatible = "ethernet-phy-id67c9.de0a";
+			reg = <5>;
+			reset-gpios = <&pio 14 1>;
+			reset-assert-us = <600>;
+			reset-deassert-us = <20000>;
+			phy-mode = "2500base-x";
+			partner_gmac = <0>;
+		};
+
+		phy6: phy@6 {
+			compatible = "ethernet-phy-id67c9.de0a";
+			reg = <6>;
+			phy-mode = "2500base-x";
+			partner_gmac = <1>;
+		};
+		phy0: ethernet-phy@0 {
+			compatible = "ethernet-phy-id03a2.9461";
```

Comment:
I prefer when you just provide the generic c22 or c45 name:
```suggestion
			compatible = "ethernet-phy-ieee802.3-c22";
```

Reply:
The `phy0: ethernet-phy@0 ` entry is not needed, it is already in the `target/linux/mediatek/files-6.6/arch/arm64/boot/dts/mediatek/mt7981.dtsi` file.